components/builder-worker/src/runner/log_pipe.rs

```diff
@@ -55,23 +63,52 @@ impl LogPipe {
     pub fn pipe(&mut self, process: &mut process::Child) {
         let mut line_count = 0;
 
+        writeln!(&self.log_file, "About to log stdout");
         if let Some(ref mut stdout) = process.stdout {
             let reader = BufReader::new(stdout);
             line_count = self.stream_lines(reader, line_count);
         }
+        writeln!(&self.log_file, "Finished logging stdout");
+        writeln!(&self.log_file, "About to log stderr");
         if let Some(ref mut stderr) = process.stderr {
             let reader = BufReader::new(stderr);
             // not capturing line_count output because we don't use it
             self.stream_lines(reader, line_count);
         }
+        writeln!(&self.log_file, "Finished logging stderr");
 
+        writeln!(
+            &self.log_file,
+            "About to tell log_forwarder that the job is complete"
+        );
         // Signal that the log is finished
         let mut complete = JobLogComplete::new();
         complete.set_job_id(self.job_id);
-        self.sock.send_str(LOG_COMPLETE, zmq::SNDMORE).unwrap();
-        self.sock
-            .send(complete.write_to_bytes().unwrap().as_slice(), 0)
-            .unwrap();
+        if let Err(e) = self.sock.send_str(LOG_COMPLETE, zmq::SNDMORE) {
+            writeln!(
+                &self.log_file,
+                "ZMQ error when sending LOG_COMPLETE: {:?}",
+                &e
+            );
+            panic!(e);
```

Comment:
Why these `panic!` calls? At this point the actual execution of the job is complete, but we apparently would have some issue closing out the log for the job. I think if we panic here, then the entire worker thread dies, and we'd then lose the entire job, even if it was successful.